### ISSUE TITLE
Installation from GitHub Mirror fails

### ISSUE THREAD

**Issue by fnogatz** [fnogatz/swivm#16]
If `SWIVM_MIRROR` is not reachable, swivm falls back to download the tarballs from GitHub (as described in https://github.com/fnogatz/swivm/issues/15#issuecomment-510234751). Apparently the package provided by GitHub is not exactly the same as from the SWI-Prolog website, so the installation fails.

**Comment by fnogatz** [comment]
@JanWielemaker wrote in https://github.com/fnogatz/swivm/issues/3#issuecomment-312099331:

> Note that the tarballs cannot be done as a github release as it contains both a (subset of) the git submodules and generated files that require extensive infrastructure to create.

This is still true -- as of now, there is no way to include the submodule's sources in the tarballs created by GitHub releases (see [GitHub Community Forum](https://github.community/t5/How-to-use-Git-and-GitHub/Create-Release-that-contains-submodule/td-p/12073)). This makes it hard for swivm to rely on the files of [SWI's GitHub releases](https://github.com/SWI-Prolog/swipl-devel/releases). Nevertheless, I was wondering if it is possible to simply download the appropriate package files again by parsing the provided `.gitmodules` to get the package's repository URL (which points to its releases list, e.g., https://github.com/SWI-Prolog/packages-chr/archive/V8.1.9.tar.gz for `chr@8.1.9`).

Turns out that although it required some `sed` magic, it is possible. I added the function `swivm_download_git_submodules` that is only called when the original download of the SWI tarball was from GitHub. It loops through the `.gitmodules` file and separately downloads the needed tarballs from the packages.

tl;dr: Downloading from GitHub is working again. The update is released as version `0.6.0`. We still need a way to let the user explicitly decide which mirror (SWI-Prolog website, or GitHub releases) to use.
